devices/Bmxx80/Bme280.cs

```diff
@@ -0,0 +1,181 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Threading;
+using Iot.Device.Bmxx80.CalibrationData;
+using Iot.Device.Bmxx80.PowerMode;
+using Iot.Device.Bmxx80.ReadResult;
+using Iot.Device.Bmxx80.Register;
+using UnitsNet;
+
+namespace Iot.Device.Bmxx80
+{
+    /// <summary>
+    /// Represents a BME280 temperature, barometric pressure and humidity sensor.
+    /// </summary>
+    [Interface("Represents a BME280 temperature, barometric pressure and humidity sensor.")]
+    public class Bme280 : Bmx280Base
+    {
+        /// <summary>
+        /// The expected chip ID of the BME280.
+        /// </summary>
+        private const byte DeviceId = 0x60;
+
+        /// <summary>
+        /// Calibration data for the <see cref="Bme680"/>.
+        /// </summary>
+        private Bme280CalibrationData _bme280Calibration;
+
+        private Sampling _humiditySampling;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Bme280"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Bme280(I2cDevice i2cDevice)
+            : base(DeviceId, i2cDevice)
+        {
+            _bme280Calibration = (Bme280CalibrationData)_calibrationData;
+            _communicationProtocol = CommunicationProtocol.I2c;
+        }
+
+        /// <summary>
+        /// Gets or sets the humidity sampling.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <see cref="Sampling"/> is set to an undefined mode.</exception>
+        [Property]
+        public Sampling HumiditySampling
+        {
+            get => _humiditySampling;
+            set
+            {
+                if (!value.Equals(Sampling.HighResolution) &&
```

Comment:
I've been more brutal than that :-) I just removed the check of IsDefined!

Reply:
Ok, but as I already converted it, I'll leave the checks in, I guess.